signalfx-connection/pom.xml

```diff
@@ -0,0 +1,68 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+  <modelVersion>4.0.0</modelVersion>
+
+  <parent>
+    <groupId>com.signalfx.public</groupId>
+    <artifactId>clients-parent</artifactId>
+    <version>1.0.30</version>
+  </parent>
+
+  <properties>
+    <!-- This prevents shading from happening in this module -->
+    <skipShaded>true</skipShaded>
+    <maven.deploy.skip>false</maven.deploy.skip>
```

Comment:
Just wondering, what's this for?

Reply:
It's wonky.  So without this (and the new setting in the parent pom), every child module/jar gets its dependencies shaded, which can result in public/external APIs having breaking changes to their interface. So a method that normally takes something like `org.apache.FooImpl` gets changed to `com.signalfx.shaded.org.apache.FooImpl` and this breaks all the callers. I couldn't find another way around that, unfortunately.

Reply:
OOF
maven magic 🪄